pants_jupyter_plugin/plugin.py

```diff
@@ -0,0 +1,403 @@
+from contextlib import contextmanager
+import asyncio
+import itertools
+import functools
+import os
+import pathlib
+import random
+import shlex
+import shutil
+import string
+import sys
+import typing
+
+from IPython.core.magic import line_magic, Magics, magics_class
+from IPython.display import display
+from IPython.display import Javascript
+from pex.pex_bootstrapper import bootstrap_pex_env
+from pex.variables import Variables
+# TODO: replace or vendor these.
+from twitter.common.contextutil import environment_as, pushd, temporary_dir
+import ipywidgets
+import nest_asyncio
+
+
+FAIL_GLYPH = '✗'
+SUCCESS_GLYPH = '✓'
+SPINNER_SEQ = '⠋⠙⠹⠸⠼⠴⠦⠧⠇⠏'
+
+
+def _scrub_import_environment(sys_modules_whitelist: typing.List[str], logger: typing.Callable):
+  """Scrubs sys.path and sys.modules to a raw state.
+
+  WARNING: This will irreversably mutate sys.path and sys.modules each time it's called.
+  """
+  pex_root = pathlib.Path(Variables().PEX_ROOT)
```

Comment:
PEX-INFO can contain a `pex_root` now so this is broken for those cases. I've outlined an idea to switch to using the Pex CLI which would solve this, but using the Pex API you'd need to:
```python
pex_info = PexInfo.from_pex(pex_path)
pex_info.update(PexInfo.from_env())
pex_root = pex_info.pex_root
```